File: frontend/src/services/accountService.js
import api from "./api";

const accountService = {

    changeUsername(newUsername) {
        return api.patch("/account/username", {
            newUsername,
        });
    },

    checkUsernameAvailability(username) {
        return api.get("/account/username/check", {
            params: { username },
        });
    },

    changePassword(currentPassword, newPassword) {
        return api.patch("/account/password", {
            currentPassword,
            newPassword,
        });
    },

    sendEmailOtp(newEmail) {
        return api.post("/account/email/otp", {
            newEmail,
        });
    },

    verifyEmail(newEmail, otp) {
        return api.patch("/account/email", {
            newEmail,
            otp,
        });
    },

    getCurrentUser() {
    return api.get("/account/me");
}
};

export default accountService;

File: src/main/java/com/sangamesh/Fitsphere/service/AccountService.java
package com.sangamesh.Fitsphere.service;

import com.sangamesh.Fitsphere.dto.account.ChangeEmailRequestDto;
import com.sangamesh.Fitsphere.dto.account.CurrentUserResponseDto;

public interface AccountService {

    void changeUsername(String newUsername);
    ChangeEmailRequestDto.UsernameAvailabilityResponseDto checkUsernameAvailability(String username);

    void changePassword(String currentPassword, String newPassword);

    void sendEmailChangeOtp(String newEmail);

    void verifyAndChangeEmail(String newEmail, String otp);

    CurrentUserResponseDto getCurrentUser();
}
